packages/office-ui-fabric-react/src/components/Modal/Modal.base.tsx

```diff
@@ -135,7 +144,7 @@ export class ModalBase extends BaseComponent<IModalProps, IDialogState> implemen
     // @temp tuatology - Will adjust this to be a panel at certain breakpoints
     if (responsiveMode! >= ResponsiveMode.small) {
       return (
-        <Layer onLayerDidMount={onLayerDidMount}>
+        <Layer {...layerProps} onLayerDidMount={onLayerDidMount}>
```

Comment:
If we are introducing the `layerProps` then we should also deprecate the `onLayerDidMount` prop of the `Modal`. Also, needs some logic in here. What happens if the user passes `onLayerDidMount` through the `layerProps`? You end up with the Modal prop `onLayerDidMount` which is `undefined` to override the spread props coming from `layerProps`.

Reply:
Introduced `_onLayerDidMount` where both functions are handled; Also marked `IModalProps.onLayerDidMount `as deprecated. Since `IDialogProps.onLayerDidMount` is already marked as deprecated nothing had to be changed there.